File: Android/app/src/androidTest/java/jp/co/riso/smartdeviceapp/SmartDeviceAppTest.java

package jp.co.riso.smartdeviceapp;

import android.content.SharedPreferences;
import androidx.preference.PreferenceManager;
import android.test.ApplicationTestCase;

public class SmartDeviceAppTest extends ApplicationTestCase<SmartDeviceApp> {

    public SmartDeviceAppTest() {
        super(SmartDeviceApp.class);
    }

    public SmartDeviceAppTest(Class<SmartDeviceApp> applicationClass) {
        super(applicationClass);
    }

    protected void setUp() throws Exception {
        super.setUp();
    }

    protected void tearDown() throws Exception {
        super.tearDown();

    }

    // ================================================================================
    // Tests - constructors
    // ================================================================================

    public void testConstructor() {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(SmartDeviceApp.getAppContext());
        SharedPreferences.Editor edit = prefs.edit();
        edit.clear();
        edit.apply();
        SmartDeviceApp smartDeviceApp = new SmartDeviceApp();
        assertNotNull(smartDeviceApp);
    }

    // ================================================================================
    // Tests - getAppContext
    // ================================================================================

    public void testGetAppContext() {
        assertNotNull(SmartDeviceApp.getAppContext());
    }
}

File: Android/app/src/androidTest/java/jp/co/riso/smartdeviceapp/SmartDeviceAppTest.kt
package jp.co.riso.smartdeviceapp

import android.test.ApplicationTestCase
import jp.co.riso.smartdeviceapp.SmartDeviceApp
import kotlin.Throws
import android.content.SharedPreferences
import androidx.preference.PreferenceManager
import androidx.test.ext.junit.runners.AndroidJUnit4
import junit.framework.TestCase
import org.junit.Test
import org.junit.runner.RunWith
import java.lang.Exception

@RunWith(AndroidJUnit4::class)
class SmartDeviceAppTest {

    // ================================================================================
    // Tests - constructors
    // ================================================================================
    @Test
    fun testConstructor() {
        val prefs = PreferenceManager.getDefaultSharedPreferences(SmartDeviceApp.getAppContext())
        val edit = prefs.edit()
        edit.clear()
        edit.apply()
        val smartDeviceApp = SmartDeviceApp()
        TestCase.assertNotNull(smartDeviceApp)
    }

    // ================================================================================
    // Tests - getAppContext
    // ================================================================================
    @Test
    fun testGetAppContext() {
        TestCase.assertNotNull(SmartDeviceApp.getAppContext())
    }
}